apps/zui/src/app/query-home/search-area/index.tsx

```diff
@@ -18,13 +22,30 @@ const Group = styled.div`
 
 export default function SearchArea() {
   const value = useSelector(Editor.getValue)
-  const query = useSelector(Current.getActiveQuery)
   const height = useSelector(Layout.getEditorHeight)
+  const runOnEnter = useSelector(Config.getRunOnEnter)
+  const dispatch = useDispatch()
+  const tabId = useTabId()
+  const onChange = (v: string) => {
+    dispatch(Editor.setValue(v))
+  }
+
+  const onKey = (e: React.KeyboardEvent) => {
+    const isEnterKey = e.key === "Enter"
+    const isModKey = e.shiftKey || cmdOrCtrl(e)
+    if (isEnterKey) {
+      if ((runOnEnter && !isModKey) || (!runOnEnter && isModKey)) {
```

Comment:
Maybe simplify this?
```suggestion
      if (runOnEnter !== isModKey) {
```

Reply:
That's too smart for me. 😅